Find a functionally equivalent code:android.appwidget.AppWidgetHostView.getRemoteContext:COMMENT
Method Modifier: private     
Comment:/**
 * Build a {@link Context} cloned into another package name, usually for the
 * purposes of reading remote resources.
 */

@@ -1,13 +1,9 @@
 {
-    // Bail if missing package name
-    final String packageName = views.getPackage();
-    if (packageName == null)
-        return mContext;
     try {
         // Return if cloned successfully, otherwise default
-        return mContext.createPackageContextAsUser(packageName, Context.CONTEXT_RESTRICTED, mUser);
+        return mContext.createApplicationContext(mInfo.providerInfo.applicationInfo, Context.CONTEXT_RESTRICTED);
     } catch (NameNotFoundException e) {
-        Log.e(TAG, "Package name " + packageName + " not found");
+        Log.e(TAG, "Package name " + mInfo.providerInfo.packageName + " not found");
         return mContext;
     }
 }

